src/engine/Service.cpp

```diff
@@ -92,6 +95,44 @@ ResultTable Service::computeResult() {
   serviceIriString.remove_suffix(1);
   ad_utility::httpUtils::Url serviceUrl{serviceIriString};
 
+  if (siblingTree_ != nullptr) {
+    // Get the result of the siblingTree, to (potentially)
+    // reduce complexity of the SERVICE query.
+    auto siblingResult = siblingTree_->getResult();
+
+    const size_t rowLimit = 100;
+    if (siblingResult->size() < rowLimit) {
+      auto siblingVariables = siblingTree_->getVariableColumns();
+
+      // Build value clause for each common variable.
+      std::string valueClauses = "{ ";
+      for (const auto& lVar : parsedServiceClause_.visibleVariables_) {
+        auto it = siblingVariables.find(lVar);
+        if (it == siblingVariables.end()) {
+          continue;
+        }
+        const auto& sVar = *it;
```

Comment:
The semantics of this are currently a bit wrong for multiple variables:

If I have the following table:
?x ?y ?z
<x1> <y1> <z1>
<x2> <y2> <z2>

Then I need the values clause

`VALUES (?x ?y ?z) { (<x1> <y1> <z1>) (<x2> <y2> <z2>)}`
with multiple variables , not three individual values clauses, otherwise I get

8 results (the cartesian product) instead of the 2 lines).

(For the exact syntax of multi varaible clauses you can look around the standard or play with the UI, the above might contain typos).